syntax = "proto3";

option java_multiple_files = true;
option objc_class_prefix = "EMS";

package emailservice;

service EmailService{
    rpc GetEmail(Query) returns (Email) {}

    rpc SayHello (HelloRequest) returns (HelloReply) {}
    // Sends another greeting
    rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message Query {
    string field = 1;
    string filter = 2;
  }

message Email {
    string email = 1;
    string name = 2;
}